src/main-app.component.ts

```diff
@@ -1,131 +1,17 @@
-import { Component, OnInit, AfterViewInit, Renderer, ViewChildren, QueryList, ElementRef} from '@angular/core';
+import { Component } from '@angular/core';
 
-import {Layers} from '../config-serializer/config-items/Layers';
-
-import {SvgKeyboardPopoverComponent} from './components/svg-keyboard-popover.component';
-import {UhkConfigurationService} from './services/uhk-configuration.service';
+import { LegacyLoaderComponent } from './components/legacy/legacy-loader.component';
+import { SideMenuComponent } from './components/sidemenu/side-menu.component';
+import { KeyMapComponent } from './components/keymap/key-map.component';
 
 @Component({
     moduleId: module.id,
     selector: 'main-app',
-    template:
-    `   <div>
-            <button #baseButton type="button" class="btn btn-default btn-lg btn-primary" (click)="selectLayer(0)">
-                Base
-            </button>
-            <button #modButton type="button" class="btn btn-default btn-lg" (click)="selectLayer(1)">
-                Mod
-            </button>
-            <button #fnButton type="button" class="btn btn-default btn-lg" (click)="selectLayer(2)">
-                Fn
-            </button>
-            <button #mouseButton type="button" class="btn btn-default btn-lg" (click)="selectLayer(3)">
-                Mouse
-            </button>
-        </div>
-        <div>
-            <svg-keyboard-popover *ngFor="let layer of layers.elements"
-                            [moduleConfig]="layer.modules.elements"
-                            (animationend)="onKeyboardAnimationEnd($event)"
-                            hidden>
-            </svg-keyboard-popover>
-        </div>
-    `,
+    templateUrl: 'src/main-app.component.html',
```

Comment:
Why don't use template with require like at the styles? Then you could use relative path.